### PR TITLE
LibWeb: Extract some IDL mixins

### PR DESCRIPTION
**Pull request by AtkinsSJ**
We had a bunch of FIXMEs in IDL files about how things should be mixins, back from before we supported those in the WrapperGenerator. So, now they are all actual mixins instead! :^)

Also added spec URLs and FIXMEs for missing methods/fields as I noticed them.

### CODE REVIEW

File: Userland/Libraries/LibWeb/DOM/ChildNode.idl [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#import <DOM/Node.idl>
+
+// https://dom.spec.whatwg.org/#childnode
+interface mixin ChildNode {
+    [CEReactions, Unscopable] undefined before((Node or DOMString)... nodes);
+    [CEReactions, Unscopable] undefined after((Node or DOMString)... nodes);
+    [CEReactions, Unscopable] undefined replaceWith((Node or DOMString)... nodes);
+    [CEReaction;s, Unscopable, ImplementedAs=remove_binding] undefined remove();
```
[the remark this answers is not in the document]
Comment (AtkinsSJ) [reply Userland/Libraries/LibWeb/DOM/ChildNode.idl:8]: Thanks! Seems like we could do with some more error checking in WrapperGenerator.

File: Userland/Libraries/LibWeb/DOM/ShadowRoot.idl [hunk 2]
```diff
@@ -1,11 +1,15 @@
 #import <DOM/DocumentFragment.idl>
+#import <DOM/InnerHTML.idl>
 
+// https://dom.spec.whatwg.org/#shadowroot
 interface ShadowRoot : DocumentFragment {
-
-    readonly attribute DOMString mode;
+    // FIXME: mode should return a ShadowRootMode
+    //readonly attribute DOMString mode;
```
Comment (linusg) [inline Userland/Libraries/LibWeb/DOM/ShadowRoot.idl:7]: `// readonly...`

**Review by linusg — COMMENTED** [review]
Nice one! +1 for adding a spec link to all mixins and interfaces :^)

---

> Makes me wonder how things were working when Document was never including GlobalEventHandlers.

Plot twist: they didn't...

File: Userland/Libraries/LibWeb/DOM/ShadowRoot.idl (hunk 2 above)
Comment (AtkinsSJ) [reply Userland/Libraries/LibWeb/DOM/ShadowRoot.idl]: Oops, that shouldn't be commented at all! Fixed. :sweat_smile:

**Pull request merged**